pkg/apis/acme/v1alpha2/types_issuer.go

```diff
@@ -317,20 +332,26 @@ type ACMEIssuerDNS01ProviderRoute53 struct {
 // ACMEIssuerDNS01ProviderAzureDNS is a structure containing the
 // configuration for Azure DNS
 type ACMEIssuerDNS01ProviderAzureDNS struct {
+	// +kubebuilder:validation:Required
 	ClientID string `json:"clientID"`
 
+	// +kubebuilder:validation:Required
 	ClientSecret cmmeta.SecretKeySelector `json:"clientSecretSecretRef"`
 
+	// +kubebuilder:validation:Required
 	SubscriptionID string `json:"subscriptionID"`
 
+	// +kubebuilder:validation:Required
 	TenantID string `json:"tenantID"`
 
+	// +kubebuilder:validation:Required
 	ResourceGroupName string `json:"resourceGroupName"`
 
 	// +optional
 	HostedZoneName string `json:"hostedZoneName,omitempty"`
 
 	// +optional
+	// +kubebuilder:validation:Enum=AzurePublicCloud;AzureChinaCloud;AzureGermanCloud;AzureUSGovernmentCloud
```

Comment:
This is already handled by validation annotations on the `type AzureDNSEnvironment string` declaration (so it is automatically re-used between uses of the type) 😄